### PR TITLE
Improved xtypes and values for system settings

### PR DESCRIPTION
**Pull request by JoshuaLuckers**
### What does it do?
Checked the system settings and changed the value to the right type and also checked if they could use another xtype like `numberfield`.

### Why is it needed?
Improve the code base.

### Related issue(s)/PR(s)
Closes issue #12284

### CODE REVIEW

**Review by alroniks — CHANGES_REQUESTED** [review]
Upgrade script required I guess.

**Comment by JoshuaLuckers** [comment]
> Upgrade script required I guess.

I agree 👍.

**Comment by alroniks** [comment]
I am wondering what if somebody added value that does not fit new xtype, maybe we should check the type and skip them, showing the warning? I think it almost impossible but still.

**Comment by JoshuaLuckers** [comment]
> I am wondering what if somebody added value that does not fit new xtype, maybe we should check the type and skip them, showing the warning? I think it almost impossible but still.

Good point. There is already a check if the system setting is using the `old_xtype`, I will try to add the check you propose another time. 

If someone else wants to add such a check: feel free to help!

**Comment by JoshuaLuckers** [comment]
@Alroniks I gave your suggestion some thought and maybe we don't need such a check. The xtype is used when you edit the system setting via the grid. Only if you edit the value it will check if it's a valid value for the defined xtype. 

A warning if the value is invalid for the new xtype would be nice but could be done in another PR. What do you think?

File: _build/data/transport.core.system_settings.php [hunk 1]
```diff
@@ -829,7 +829,7 @@
 $settings['link_tag_scheme']->fromArray(array (
   'key' => 'link_tag_scheme',
   'value' => -1,
-  'xtype' => 'textfield',
+  'xtype' => 'numberfield',
```
Comment (Mark-H) [inline _build/data/transport.core.system_settings.php]: This should be text, can also be `full`, `https`, `abs`, etc.

File: setup/includes/upgrades/common/3.0.0-update-xtypes-system-settings.php [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    ],
+    [
+        'key' => 'log_level',
+        'old_xtype' => 'textfield',
+        'new_xtype' => 'numberfield',
+    ],
+    [
+        'key' => 'link_tag_scheme',
+        'old_xtype' => 'textfield',
+        'new_xtype' => 'numberfield',
```
Comment (Mark-H) [inline setup/includes/upgrades/common/3.0.0-update-xtypes-system-settings.php]: This one should be removed per other comment.

**Review by Mark-H — CHANGES_REQUESTED** [review]
Spotted one incorrect change, other than that, nice work.

**Review by Mark-H — APPROVED** [review]

**Review by Mark-H — APPROVED** [review]

**Pull request merged**